pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -914,8 +914,12 @@ func buildLocalityLbEndpointsFromShards(
 				localityEpMap[ep.Locality] = locLbEps
 			}
 			if ep.EnvoyEndpoint == nil {
+				tlsLabel := ep.TLSMode
+				if !push.BetaPolicyAbleAcceptMTLS(ep.Labels) {
```

Comment:
It's may be better to create the (beta) applier and access the consolidate policy instead of implement this function in push context . You may cache that applier (or the consolidate mTLS mode) per unquie labels set for performance.

Reply:
right. i will reuse as much as possible from the existing resolution.

on the caching part, the would that also be applied to the inbound filter chain construction? a.k.a the PR you currently working on? or it's because LDS code path are invoked once per push context construction, and we lack the data structure that can be shared among multiple proxy's LDS code path?

Reply:
hmm, after closer look, i think caching is kind of inevitable.

the policy applier construction requires the server workload labels, and then do some calculation. we can't offer to do policy applier construction per endpoint. a hashmap from unique label set to the value of mtls capable or not would help.

i think i can put a label uid, hash from the label `map[string] string` to use that as key.

Reply:
Can we start with simple cache for the loop in eds first? I.e if cache found, use it. Otherwise, create new applier and compute the mTLS mode (then store to cache). For now, the cost of new applier is not 0, but I think it's bearable.

For reuse the applier, it would be better if the authn policy applier factory take care of that. Putting them in the push_context is too convoluted. Plus, cache only the mTLS mode might not be enough for other use cases.

Reply:
yeah, creating applier once + map for caching is what i'm thinking of.